pandas/_libs/window/aggregations.pyx

```diff
@@ -1271,7 +1271,7 @@ def roll_apply(object obj,
         Py_ssize_t i, s, e, N = len(start), n = len(obj)
 
     if n == 0:
-        return obj
+        return np.array([], dtype=np.float64)
```

Comment:
Technically this could return an empty Series, but not sure if that will be impactful downstream.

This behavior is more consistent with the other functions when the input is empty, so we'll see what the tests say.